### PR TITLE
Move some offerToReceive tests to webrtc/legacy

### PR DESCRIPTION
**Pull request by youennf**
Moved RTCPeerConnection-createOffer-offerToReceive.html to webrtc/legacy.
Moved some tests from RTCPeerConnection-createOffer.html to webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html.

### CODE REVIEW

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html [hunk 1]
```diff
@@ -0,0 +1,274 @@
+<!doctype html>
```
Comment (alvestrand) [inline webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: Can you add a README file to the legacy/ directory saying that these tests exercise code from section 4.4.3 "Legacy extensions", APIs that are optional to implement?

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        offerToReceiveVideo: true
+      }).then(offer2 => {
+        assert_equals(countAudioLine(offer2.sdp), 1,
+          'Expect audio line to remain in created offer');
+
+        assert_equals(countVideoLine(offer2.sdp), 1,
+          'Expect video line to be found in create offer');
+      })
+    });
+  }, 'createOffer() with offerToReceiveAudio:true then offerToReceiveVideo:true should have result offer with both audio and video line');
```
Comment (alvestrand) [inline webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: language nit: s/ then / and /

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+
+    return pc.createOffer({
+      offerToReceiveAudio: true,
+      offerToReceiveVideo: false
+    }).then(offer1 => {
+      assert_equals(countAudioLine(offer1.sdp), 1,
+        'Expect audio line to be found in created offer');
+
+      assert_equals(countVideoLine(offer1.sdp), 0,
+        'Expect video line to not found in create offer');
```
Comment (alvestrand) [inline webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: "to not found" -> "to not be found"

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    }, `offerToReceive${capsKind} set to false with a track should create a "sendonly" transceiver`);
+
+    promise_test(t => {
+      const pc = new RTCPeerConnection();
+
+      t.add_cleanup(() => pc.close());
+
+      pc.addTransceiver(kind, {direction: 'recvonly'});
+
+      return pc.createOffer(offerToReceiveFalse)
```
Comment (alvestrand) [inline webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: Should we bite the bullet to change these tests into async/await code?
Or do you just want to get it over with?

**Review by alvestrand — COMMENTED** [review]
Some nits....

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html (hunk 2 above)
Comment (youennf) [reply webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: Updated the title with commas

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html (hunk 1 above)
Comment (youennf) [reply webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: OK

File: webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html (hunk 4 above)
Comment (youennf) [reply webrtc/legacy/RTCPeerConnection-createOffer-offerToReceive.html]: I prefer just copying the tests if possible for now.

**Review by alvestrand — APPROVED** [review]

**Pull request merged**